### PR TITLE
Houston ToS, Privacy Policy, Cookie Policy, and Cookie Prompt Modal

### PR DESCRIPTION
**Pull request by sahilakos**

### CODE REVIEW

**Comment by josh-padnick** [comment]
1. Visit http://localhost:4000
1. Visit http://localhost:4000/infrastructure-as-code-library/
1. Click "OK" on cookie prompt
1. Visit http://localhost:4000/how-it-works/.

The cookie prompt still shows up. Tested in Chrome and Firefox.

**Comment by josh-padnick** [comment]
Cookie prompt looks great on mobile.

**Comment by josh-padnick** [comment]
On the Cookie Policy page, could you change:



_The following describes the types of cookies we use with our Services. All third parties listed below set third-party cookies with our Services, but please note that the list is not an exhaustive list at all times of all third parties who set third-party cookies with our Services. To opt out of any cookies, see the section Opting out of cookies._

to:

_The following describes the types of cookies we use with our Services. To opt out of any cookies, see the section [Opting out of cookies](#). All third parties listed below set third-party cookies with our Services, but **please note that this list is not an exhaustive list at all times of all third parties who set third-party cookies with our Services.**_

**Comment by josh-padnick** [comment]
Strange link formatting:

image

**Comment by sahilakos** [comment]
Yea, I meant to leave a note about the "Choices About ....", that line should link to different parts for the privacy policy for use and disclose. Right?

And, about cookie prompt, I'll have to check why it's not functioning once it's accepted. I had tested it initially. Will fix that.

I'll make the other content updates as well. Will be back to work on this later tonight so you have a little time to catch any other issues.

**Comment by josh-padnick** [comment]
Ah, you know the "Choices About..." should really just be [How can you impact our collection and use of your data?](http://localhost:4000/privacy-policy/#how-can-you-impact-our-collection-and-use-of-your-data). That was a typo in the original. 

Sounds good on next steps. I realize this project was a lot of meticulous text and not a lot of fun development. Thanks for powering through it. We're pretty close to done!

**Comment by sahilakos** [comment]
@josh-padnick The cookie prompt should only appear once. 

To Test:

- Chrome: Inspect element > Applications Tab > Cookies list in left nav. And you should see a GruntyCookie in the list persistent through the session. https://cl.ly/059cc31ea8e2
- Firefox: Inspect Element > Storage Tab > Cookies list in left nav. https://cl.ly/024d5471c3e3

Also updated the text & link changes on cookie policy and privacy policy.

File: assets/js/main.js [hunk 1]
```diff
@@ -23,11 +23,11 @@ function serialize(a,b){"object"!=typeof b?b={hash:!!b}:void 0===b.hash&&(b.hash
 
 /**
  * jQuery.query - Query String Modification and Creation for jQuery
- * Written by Blair Mitchelmore (blair DOT mitchelmore AT gmail DOT com)
+ * Written by Blair MitchcookieModalre (blair DOT mitchcookieModalre AT gmail DOT com)
  * Licensed under the WTFPL (http://sam.zoy.org/wtfpl/).
  * Date: 2009/8/13
  *
- * @author Blair Mitchelmore
+ * @author Blair MitchcookieModalre
```
Comment (brikis98) [inline assets/js/main.js]: Looks like accidental search/replace?

File: assets/js/main.js [hunk 2]
```diff
@@ -370,4 +370,58 @@ $(function() {
...
+
+
+  //check if cookie already exists
+  if(document.cookie){
+    var cookieString = document.cookie;
+    var cookieList = cookieString.split(";");
+    // if cookie named GruntyCookie is found, return
+    for(x = 0; x < cookieList.length; x++){
+      if (cookieList[x].indexOf("GruntyCookie") != -1){
+        console.log(cookieList[x].indexOf("GruntyCookie"));
```
Comment (brikis98) [inline assets/js/main.js]: Remove?

File: assets/js/main.js [hunk 3]
```diff
@@ -370,4 +370,58 @@ $(function() {
...
+/* Add cookie prompt functionality */
+(function(){
+  var linkText = "cookie policy";
+  var linkUrl = "/cookie-policy/";
+  var dismissBtn = "OK";
+  var msg = "By using this website you agree to our " + linkText.link(linkUrl);
+
+
+  //check if cookie already exists
+  if(document.cookie){
```
Comment (brikis98) [inline assets/js/main.js]: Might be cleaner to add a reusable `getCookie(name)` function: https://stackoverflow.com/a/25490531/483528

File: assets/js/main.js [hunk 4]
```diff
@@ -370,4 +370,58 @@ $(function() {
...
+  cookieModalCclose.setAttribute("class", "btn btn-primary");
+  cookieModalCclose.appendChild(cookieModalCloseText);
+  cookieModalCclose.addEventListener("click", closeCookie, false);
+
+
+  //add to DOM
+  //cookieModalCp.appendChild(cookieModalCText);
+  cookieModalC.appendChild(cookieModalCp);
+  cookieModalC.appendChild(cookieModalCclose);
+  docRoot.appendChild(cookieModalC);
```
Comment (brikis98) [inline assets/js/main.js]: Why create this all in JS? Just create the modal in the HTML, but set it to hidden by default. Then use JS to show/hide that modal.

File: assets/js/main.js [hunk 5]
```diff
@@ -370,4 +370,58 @@ $(function() {
...
+
+  cookieModalC.classList.add("cookieModalCBeginAnimate");
+
+  function closeCookie(){
+    var cookieExpire = new Date();
+    cookieExpire.setFullYear(cookieExpire.getFullYear() +2);
+    document.cookie="GruntyCookie=1; path=/; expires=" + cookieExpire.toGMTString() + ";";
```
Comment (brikis98) [inline assets/js/main.js]: A reusable `setCookie()` method would be cleaner. Here's another example: https://stackoverflow.com/a/24103596/483528

File: assets/js/main.js (hunk 1 above)
Comment (sahilakos) [reply assets/js/main.js]: Yea, I was replacing the name of cookie modal "elmo" to just "cookieModal". I'll fix it.

File: assets/js/main.js (hunk 2 above)
Comment (sahilakos) [reply assets/js/main.js]: Ah, yes, forgot in rush.

**Comment by josh-padnick** [comment]
@sahilakos Any chance we can finish this up today?

**Comment by sahilakos** [comment]
@josh-padnick Yes, in my queue for today.

**Comment by josh-padnick** [comment]
Made some final updates in 4e06065, and we are officially ready to launch! Thank you so much for your help @sahilakos 🍺

**Pull request merged**